receiver/kafkareceiver/kafka_receiver.go

```diff
@@ -446,7 +447,10 @@ func (c *tracesConsumerGroupHandler) ConsumeClaim(session sarama.ConsumerGroupSe
 			}
 
 			ctx := c.obsrecv.StartTracesOp(session.Context())
-			statsTags := []tag.Mutator{tag.Upsert(tagInstanceName, c.id.String())}
+			statsTags := []tag.Mutator{
+				tag.Upsert(tagInstanceName, c.id.String()),
+				tag.Upsert(tagPartition, strconv.Itoa(int(claim.Partition()))),
```

Comment:
I would prefer to have this an opt in tag due to potential cardinality increase in can bring

Reply:
It's true that you'd get increased cardinality (instead of 3 metrics per receiver, you'd have 3*topic-partition-count), but right now, the metrics are just wrong. If you take a look at the chart in the [issue](https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30177) you'll see the counter fluctuating according to the latest partition to be read.  This plays havoc with prometheus `rate` calculations, because it ignores counter values decreasing (they aren't allowed to decrease) but it counts all increases.

Imagine this scenario:
* topic has 2 partitions
* partition 1 offset: 0
* partition 2 offset: 1000
* both partitions have 1 span per 2 prometheus scrapes (each alternating)

In this case, prometheus will see the counter values: [1, 1000, 2, 1001, 3, 1003]

So in this case, we have 0.5 spans per scrape, but the rate function will return ~3000 spans per scrape.

I pulled these graphs just now, and they're from the same collector, one being sarama offsets and the other being processed messages. Queries were
```
sum(rate(otelcol_kafka_receiver_current_offset{... some stuff ...}[3m]))
sum(rate(otelcol_receiver_accepted_spans{... same stuff ...}[3m]))
```
![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/216748/75534a3c-9d5f-48f0-88e2-7a208f4da23c)
![image](https://github.com/open-telemetry/opentelemetry-collector-contrib/assets/216748/a9cda1c7-09dc-44e7-a956-95b325ed7d9a)

Reply:
I was thinking about how we could remove the cardinality and just roll everything into one single metric (e.g. by just returning the sum of all our offset deltas since we start up) but I think in at least some cases it would violate Kafka's semantic modek.  e.g. In my head, I could justify the `receiver_offset_lag` metric being aggregated e.g. `warning, there is a general rising backlog of unprocessed spans`, but `current_offset` is 100% tied to a partition and not a topic.

Reply:
That makes sense to me, if it doesn't make sense to provide a less detailed metric, then an alternative metric makes the most sense to me.

Reply:
There is a partition-agnostic metric by the way - `otelcol_kafka_receiver_messages`